### ISSUE TITLE
Pipx installation of extras impossible due to spaces

### ISSUE THREAD

**Issue by lukasjuhrich** [joshuabach/gnucash-web#44]
## What happens
Trying to install gnucash-web with the postgres extras, the following happened:
```shell-session
➜  ~ pipx install 'gnucash-web[PostgreSQL backend]' --pip-args 'Flask<2.3' --verbose
pipx >(setup:801): pipx version is 1.2.0
pipx >(setup:802): Default python interpreter is '/usr/bin/python'
Unable to parse package spec: gnucash-web[PostgreSQL backend]
```

## Why it happens
The extras are defined here:
https://github.com/joshuabach/gnucash-web/blob/085a63dc9ce5d772a081f8936392567576718105/setup.py#L58-L61

Pipx uses [`packaging.Requirements`](https://packaging.pypa.io/en/stable/requirements.html) to parse the spec, see
[the source](https://github.com/pypa/pipx/blob/d8ff1890e2eeb5e7bb6f27b3d326df879f699e90/src/pipx/package_specifier.py#L48-L109). What a named requirement should look like is determined by [PEP508](https://peps.python.org/pep-0508/#grammar):
```
identifier_end = letterOrDigit | (('-' | '_' | '.' )* letterOrDigit)
identifier    = letterOrDigit identifier_end*
name          = identifier
extras_list   = identifier (wsp* ',' wsp* identifier)*
extras        = '[' wsp* extras_list? wsp* ']'

name_req      = name wsp* extras? wsp* versionspec? wsp* quoted_marker?
```
In other words: spaces are not allowed in the `extras` part..

## Fix
Just rename the extras requirements to `postgres` and `mysql`.

**Comment by joshuabach** [comment]
Huh, I've never used pipx and it works fine with pip and setup.py. But it seems it is indeed a spec violation, so I'll fix this as soon as I can.

**Comment by lukasjuhrich** [comment]
Thanks for the quick reply!

Just FYI, pip v23.2 (setuptools 68.0.0) on python 3.11.3 seems to fail as well, both with command line and `requirements.txt` file:

<details>
<summary><i>(shell output)</i></summary>

```
(test) ➜  fin python --version
Python 3.11.3
(test) ➜  fin pip --version   
pip 23.2 from /home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip (python 3.11)
(test) ➜  fin pip install 'gnucash-web[PostgreSQL backend]'
ERROR: Exception:
Traceback (most recent call last):
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_vendor/packaging/requirements.py", line 102, in __init__
    req = REQUIREMENT.parseString(requirement_string)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_vendor/pyparsing/util.py", line 256, in _inner
    return fn(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_vendor/pyparsing/core.py", line 1190, in parse_string
    raise exc.with_traceback(None)
pip._vendor.pyparsing.exceptions.ParseException: Expected string_end, found '['  (at char 11), (line:1, col:12)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_internal/cli/base_command.py", line 180, in exc_logging_wrapper
    status = run_func(*args)
             ^^^^^^^^^^^^^^^
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_internal/cli/req_command.py", line 248, in wrapper
    return func(self, options, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_internal/commands/install.py", line 342, in run
    reqs = self.get_requirements(args, options, finder, session)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_internal/cli/req_command.py", line 411, in get_requirements
    req_to_add = install_req_from_line(
                 ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_internal/req/constructors.py", line 395, in install_req_from_line
    parts = parse_req_from_line(name, line_source)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_internal/req/constructors.py", line 332, in parse_req_from_line
    extras = convert_extras(extras_as_string)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_internal/req/constructors.py", line 57, in convert_extras
    return get_requirement("placeholder" + extras.lower()).extras
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_internal/utils/packaging.py", line 45, in get_requirement
    return Requirement(req_string)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukas/.virtualenvs/test/lib/python3.11/site-packages/pip/_vendor/packaging/requirements.py", line 104, in __init__
    raise InvalidRequirement(
pip._vendor.packaging.requirements.InvalidRequirement: Parse error at "'[postgre'": Expected string_end
```
</details>

So this is not necessarily specific to pipx.
